### PR TITLE
Resolve debounce timing bug

### PR DESCRIPTION
**Pull request by zanderz**
Bug was that the first user in the search results list would be selected immediately when hitting space, tab or comma, though the search was still being debounced, which lasts one second, causing the wrong result to be pilled.

Now, hitting those keys while typing will short-circuit the debuncer and not cause selection. After debouncing has elapsed, they will do the selection as before. In other words, if you quickly type "m-a-x-comma", max will be at the top of the list but not pilled. Hitting comma again will then pill it.

@keybase/react-hackers

### CODE REVIEW

**Comment by chrisnojima** [comment]
i don't think we should use a timing based solution to this. that'll be super flakey. we have all the info we need in the store already right? 
you type 'max,' and if the results we show aren't cause of 'max' then we shouldn't do anything. you just have to (if we don't already) store which keyword we used to actually do the search and set it when we get the results back
so

i type 'ma' then 'x'
i get the 'ma' results back. i write that in the store searchedTerm: 'ma' results: [...]
then if i get the comma, i see i'm trying to comma 'max' but we're showing the results for 'ma' so we ignore

**Comment by gabriel** [comment]
I think if you echo back the input with the results (response) you can ignore results when the input doesn't match it..

-> search 'ma'
-> search 'x'
<- result for 'ma' (ignored)
<- result for 'x'

**Comment by zanderz** [comment]
There is already a timing based stage not reflected in the store/state: the debouncer, which is the main part of this bug in the first place. It goes like:
type 'M' -> type 'A' ... type X -> debouncing expires (1 sec) -> call search -> check cache (miss) -> set pending in store -> call API -> register results -> set pending off in store
   _or_ check cache (hit) -> register results

So, the debounce delay is not registered in the store/state, nor is the search pending if the cache has it.
Since the debouncer object doesn't export the info we need about what is pending, this PR attempts to mirror it.

I slightly disagree about strict matching: if you type "m" and see "mikem" highlighted at the top, hitting enter should pick it.

**Comment by chrisnojima** [comment]
But your local state does know the value of the input so it knows you're typed 'ma' and not 'm' even if thats all redux knows.

We shouldn't be doing any local matching, we should be doing the bookkeeping of which results came from which query

File: shared/searchv3/helpers.js [hunk 1]
```diff
@@ -68,21 +68,33 @@ const onChangeSelectedSearchResultHoc = compose(
...
-        props.search(nextText, props.selectedService)
-      } else {
-        props._searchDebounced(nextText, props.selectedService)
-      }
-    },
+  withHandlers(() => {
+    let lastSearchTerm
+    return {
+      onAddSelectedUser: (props: OwnPropsWithSearchDebounced) => () => {
+        props._searchDebounced.flush()
```
Comment (chrisnojima) [inline shared/searchv3/helpers.js]: why run the flush? does it sync update the props?

**Pull request merged**